source/server/admin/prometheus_stats.cc

```diff
@@ -184,10 +193,17 @@ std::string PrometheusStatsFormatter::formattedTags(const std::vector<Stats::Tag
 std::string PrometheusStatsFormatter::metricName(const std::string& extracted_name) {
   std::string sanitized_name = sanitizeName(extracted_name);
 
-  absl::string_view prom_namespace{sanitized_name};
-  prom_namespace = prom_namespace.substr(0, prom_namespace.find_first_of('_'));
-
-  if (prometheusNamespaces().contains(prom_namespace)) {
+  absl::string_view view{sanitized_name};
+  const auto first_pos = sanitized_name.find_first_of('_');
+  view = view.substr(0, first_pos);
+  if (Stats::Utility::customStatNamespaceRegistered(view)) {
```

Comment:
I think we should make a utility class to hold the registered namespaces and do the sorts of string-transformations you want as a method of that class. This way, the same logic could be shared by all sinks.

Also I think it might be worth discussing whether to be more restricted about how these namespaces work, and do this transformation in symbol-table space rather than in string space.

Reply:
I guess doing in symbol-table space would be much more complex impl vs efficiency, though I think the current string space impl is not that bad - custom metrics incur one additional copy of name (trimming the custom stat namespace from the original), and operations on  the other native metrics are the same as before.